include/cell.tcc

```diff
@@ -951,3 +951,118 @@ Eigen::VectorXd mpm::Cell<Tdim>::interpolate_nodal_acceleration(
 
   return acceleration;
 }
+
+//! Assign velocity constraint
+//! Constrain directions can take values between 0 and Dim-1
+template <unsigned Tdim>
+bool mpm::Cell<Tdim>::assign_cell_velocity_constraint(unsigned face_id,
+                                                      unsigned dir,
+                                                      double velocity) {
+  bool status = true;
+  try {
+    //! Constrain directions can take values between 0 and Dim * Nphases - 1
+    if (dir >= 0 && dir < Tdim) {
+      if (velocity_constraints_.find(face_id) == velocity_constraints_.end())
```

Comment:
It is stored as a map, you don't need to do use find.

Reply:
Yes, but I want it to return `end()` if it is not found. So what this line and the next do are to check whether we have `face_id` for the `first`. If we do, simply `emplace_back` but if we don't, make a pair then emplace.

Reply:
I see, that would be quicker. Can we do it later though and leave it for now?

Reply:
Sure, please place a  `// TODO` comment here, and add an issue to come back later, when this PR has been merged.

Reply:
Could you fix this in this PR? Any particular reason why you want to do it later?

Reply:
Yes, will do it in this PR too!